### PR TITLE
pwx-36986 | test: add failback integration tests for new dr workflow

### PR DESCRIPTION
**Pull request by strivedi-px**
**What type of PR is this?**
>integration-test

**What this PR does / why we need it**:
Addresses PWX-36986. Adds tests for the following cases:
1. [[Failback] Perform failback of an app whose number of replicas have been modified in secondary after failover. ](https://portworx.testrail.net/index.php?/cases/view/257160)

2. [[Failback] Perform failback of a subset of namespaces which have been activated after failover.](https://portworx.testrail.net/index.php?/cases/view/257159)

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No

**Test runs**
- Pipeline run: https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/Stork%2Fapplication-backup-24-1/detail/application-backup-24-1/2242/pipeline
- Local run:
Screenshot from 2024-05-17 02-40-43

### CODE REVIEW

File: test/integration_test/migration_dr_actions_test.go [hunk 1]
```diff
@@ -644,6 +646,577 @@ func testDRActionFailbackMonthlyScheduleTest(t *testing.T) {
...
+	log.FailOnError(t, err, "error retrieving deployments from %s namespace", mysqlNamespace)
+	Dash.VerifyFatal(t, len(sourceDeployments.Items), 1, fmt.Sprintf("Expected 1 deployment in source in %s namespace", mysqlNamespace))
+	Dash.VerifyFatal(t, *sourceDeployments.Items[0].Spec.Replicas, 1, fmt.Sprintf("Expected 0 replica in source in deployment in %s namespace", mysqlNamespace))
+
+	sourceStatefulsets, err = apps.Instance().ListStatefulSets(elasticsearchNamespace, metav1.ListOptions{})
+	log.FailOnError(t, err, "error retrieving statefulsets from %s namespace", elasticsearchNamespace)
+	Dash.VerifyFatal(t, len(sourceStatefulsets.Items), 1, fmt.Sprintf("Expected 1 statefulset in source in %s namespace", elasticsearchNamespace))
+	Dash.VerifyFatal(t, *sourceStatefulsets.Items[0].Spec.Replicas, 0, fmt.Sprintf("Expected 0 replica in source in sts in %s namespace", elasticsearchNamespace))
+
+	// Failover #2 mysql ns
```
Comment (diptiranjanpx) [inline test/integration_test/migration_dr_actions_test.go:1051]: invalid comment.

File: test/integration_test/migration_dr_actions_test.go [hunk 2]
```diff
@@ -644,6 +646,577 @@ func testDRActionFailbackMonthlyScheduleTest(t *testing.T) {
...
+	// Create schedulePolicies using storkCtl.
+	cmd = storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	for schedulePolicyName, customArgs := range reverseSchedulePolicyArgs {
+		cmdArgs := []string{"create", "schedulepolicy", schedulePolicyName, "--kubeconfig", destinationKubeConfigPath}
+		executeStorkCtlCommand(t, cmd, cmdArgs, customArgs)
+	}
+	log.Info("created reverse schedule policy")
+
+	// Create migration schedule.
+	reverseMigrationScheduleName := "reverse-failback-modified-secondary-replicas"
```
Comment (diptiranjanpx) [inline test/integration_test/migration_dr_actions_test.go:1116]: change the name of this schedule as we are not modifying replicas

**Review by diptiranjanpx — APPROVED** [review]
LGTM after addressing minor comments.

File: test/integration_test/migration_dr_actions_test.go (hunk 1 above)
Comment (strivedi-px) [reply test/integration_test/migration_dr_actions_test.go]: Addressed in the commit bffa82b.

File: test/integration_test/migration_dr_actions_test.go (hunk 2 above)
Comment (strivedi-px) [reply test/integration_test/migration_dr_actions_test.go]: Done.

**Pull request merged**